assets/js/src/core/modules/app/app-loader.tsx

```diff
@@ -31,17 +34,26 @@ export interface IAppLoaderProps {
 
 export const AppLoader = (props: IAppLoaderProps): React.JSX.Element => {
   const dispatch = useAppDispatch()
-  const [translations] = useTranslationGetCollectionMutation()
   const { i18n } = useTranslation()
+
   const [isLoading, setIsLoading] = useState(true)
+  const [errorData, setErrorData] = useState<FetchBaseQueryError | SerializedError | null | undefined>(null)
```

Comment:
I think this is still a bit more work than needed.
That would be the recommendation from my side. 

``` typescript
class APIError extends Error {
  // .... Extend it until it can handle the Errors from RTK-Query
}

const MyComponent = () => {
  const { report } = useErrorHandler();
  const { isLoading, isError, error} = useApiHook() // <= some GET request example
  const [ 
    fetchMutation, 
    {  isError, error } // only use these when you want to force a rendering error, so that it get catched by error boundary
  ] // some Post request / mutation example

  // Should be catched by error boundary. Because it's a error during the render of the react app
  if (isError) {
    report(new APIError(error))
  }
  
  const onClick = () => {
    fetchMutation()
      .catch(() => {
        // need to be catched by ErrorHandler since the error boundary can't detect that error, because this is not causing rendering issues
        report(new APIError(error))
      })
  }
  
  return (
    <Button onClick={onClick}>My button</Button>
  )
}
```

Reply:
I agree on that approach in general but instead of `report(new APIError(error))` a small helper which does this based on the response would be an advantage in my opinion.

Reply:
@vin0401 I like your approach, already updated it based on your suggestion ☺️